src/client/app/actions/monaco.ts

```diff
@@ -27,7 +27,7 @@ export class MonacoActionTypes {
 export class ChangeTabAction implements Action {
     readonly type = MonacoActionTypes.CHANGE_TAB;
 
-    constructor(public payload: string, public language: string) { }
+    constructor(public payload: { name: string, language: string }) {}
```

Comment:
I don't know why, but when the constructor was declared as payload and then another parameter, the other parameter was getting lost on the receiving end.  It seems as though it wasn't getting serialized, or something like that.  I then noticed that all of your other constructors always had a single parameter to them -- hence I merged it in to a single object, and things started working again!  Without it, language was always null